feat: Add RTL support

limitless-dev opened the pull request:
This PR introduces RTL (Right-to-Left) support for the `Toaster` component.

![direction](https://github.com/emilkowalski/sonner/assets/24194621/0874e461-c882-441a-b1b0-1aa8c18327a7)


**Related Issues**
- #108

#### Changes:

1. RTL adjustments for SVG icons within the `Toaster`.
2. Corrected Acton button margins for RTL layout in `Toaster`.
3. Support RTL through `<html dir='rtl'>`
4. Support RTL through `<Toaster dir='rtl' />`
5. dir attributes are `rtl`, `ltr`, `auto`
6. Headless example changed to support rtl (through Toaster or html)

**Priority of direction (`dir`) :**
1. Toaster
2. html
3. if both not set, default to `ltr`

```jsx
<Toaster dir="ltr" />
<Toaster dir="auto" /> // html direction or ltr
<Toaster dir="rtl" />
<Toaster /> // html direction or ltr
```

 ```tsx
<html dir="rtl">
<html dir="ltr">
```

I know we can hack this with CSS, but it's so much nicer to have it built-in. Saves us from redoing it in every project.

joaom00 reviewed (commented):
Hey @limitless-dev, thanks for your contribution. Just some suggestions

- I think we can refactor to use [logical properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_logical_properties_and_values) here.

- Some alignments are incorrect (I think this should be fixed using logical properties):
![image](https://github.com/emilkowalski/sonner/assets/48808846/0b2146ac-1a2f-438d-a0fb-417b535f9afb)

- Add `dir` type to `Toaster` props

limitless-dev commented:
Hi @joaom00 - Thanks for taking the time to review!

> I think we can refactor to use logical properties
> Some alignments are incorrect

I just checked again and it appears to work on Firefox but not Chrome!  
I'll see what I can do.

 >   Add `dir` type to `Toaster` props

Actually, that was my first approach but I realized I'll need to pass `dir='rtl'` to both:
1. `Toaster`: since I assume will be defaulting the direction prop as LTR. 
2. `html`: since that's usually the default way of enabling RTL

Beside, that can be done through the `style` prop:
```tsx
<Toaster
  style={{ direction: 'rtl' }}
/>
```

So I focused on the styling issues for SVG and action button.

limitless-dev commented:
Fixed the issue, RTL should now function correctly (limited to HTML direction only).
Please let me know if having the `dir` as prop for `Toaster` is a preferable approach. Thank you.

joaom00 reviewed (commented):
> Please let me know if having the dir as prop for Toaster is a preferable approach

I think we can keep on `html` only.

One thing missing is inverting `[data-close-button]` as well and, if you don't mind, can you invert the position of the dismiss button in the `Headless` example? Thanks

Something I'm in doubt is whether we should invert the position of the toast too :thinking:

limitless-dev commented:
Hi @joaom00 , thanks for your comments.
I have added the `dir` as prop to `Toaster` and revamped the styling for readability.
Not sure why the build failed on Vercel but the preview works, so please test and let me know.

**Priority of `dir`:**
1. Toaster
2. html
3. if both not set, default to `ltr`

I skipped support for `auto` as `dir` prop, because I'm not sure if the following will work with NestJS:
```tsx
dir={dir === 'auto' ? window.getComputedStyle(document.documentElement).direction : dir}
```
Maybe we need `useEffect` for it?

**Changes**
- `dir` support for Toaster (ltr and rtl only)
- styling icon and button with CSS variables based on directions
- `Headless` example now supports `rtl` (through Toaster or html) 

> and in the website add an option to enable the RTL mode

Once we finalize this as stable, I'll add an example 👍

emilkowalski commented on website/src/components/Direction/index.tsx:
```diff
@@ -0,0 +1,49 @@
+
+export type Direction = (typeof directions)[number];
+
+export const Direction = ({
```
Let's remove this and add it to the Readme instead

emilkowalski commented:
This is awesome, left one comment. Can we add tests for this as well?

limitless-dev commented:
> This is awesome, left one comment. Can we add tests for this as well?

Hi @emilkowalski, thank you for your review.
- Removed `Direction` example from website
- Added `Direction` guide to README
- Added 3 tests for `dir` prop
  - toaster's dir prop is reflected correctly
  - toaster respects the HTML's dir attribute
  - toaster respects its own dir attribute over HTML's

emilkowalski commented:
Amazing, thank you so much for this
